File: Day_010/kadane-algorithm.py
class Solution:
    def maxSubarraySum(self, arr):
        a ,b = arr[0], arr[0]
        
        for i in range (1 ,len(arr)):
            a = max(arr[i], arr[i] + a)
            b = max(a,b)
            
        return b

File: Day_010/kadane-algorithm.java
class Solution {
    int maxSubarraySum(int[] arr) {
        int s = arr[0] ,e =  arr[0];
        
        for (int i = 1; i < arr.length; i++){
            s = Math.max(arr[i], arr[i] + s);
            e = Math.max(s,e);
        }
        
        return e;
    }
}
